### PR TITLE
Steve dev

### PR DESCRIPTION
**Pull request by mbouadeus**
This pull request mainly adds some refactoring of my previous code, the upload image feature when posting comments using blobstore, and the functions I started implementing for the page translation feature. Sorry for the long pull request. This is code since Friday.

[Add language selection menu] - https://screenshot.googleplex.com/vNFgOUvTYCd.png
[Change background modal to work with settings dropdown] - https://screenshot.googleplex.com/1rsteoiOO99.png
[Add upload image feature to comment form] - https://screenshot.googleplex.com/ZkmJsxd8UVd.png
https://screenshot.googleplex.com/crkdLg4yJ1C.png

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/data/CommentUtility.java [hunk 1]
```diff
@@ -104,31 +122,29 @@ public static void postComment(String collectionID, HttpServletRequest request)
     commentEntity.setProperty("name", name);
     commentEntity.setProperty("comment", comment);
     commentEntity.setProperty("timestamp", timestamp);
+
+    // Save comment image if it exists
+    String imageUrl = getUploadedFileUrl(request, "image");
+
+    System.out.println(imageUrl);
```
Comment (pengxianzhe) [inline portfolio/src/main/java/com/google/sps/data/CommentUtility.java:129]: This line looks like test code. Please remove

File: portfolio/src/main/java/com/google/sps/data/CommentUtility.java [hunk 2]
```diff
@@ -104,31 +122,29 @@ public static void postComment(String collectionID, HttpServletRequest request)
...
+    if (imageUrl != null) {
```
Comment (pengxianzhe) [inline portfolio/src/main/java/com/google/sps/data/CommentUtility.java:130]: Can you set null to a property? I have a feeling that this if statement is not necessary, but I am not sure.

File: portfolio/src/main/java/com/google/sps/data/GeneralUtility.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  *         was not specified by the client
+  */
+  public static String getParameter(HttpServletRequest request, String name, String defaultValue) {
+    String value = request.getParameter(name);
+    if (value == null) {
+      return defaultValue;
+    }
+    return value;
+  }
+}
```
Comment (isaacwdavis) [inline portfolio/src/main/java/com/google/sps/data/GeneralUtility.java:20]: nit: no new line

**Review by isaacwdavis — APPROVED** [review]

File: portfolio/src/main/java/com/google/sps/data/GeneralUtility.java (hunk 3 above)
Comment (pengxianzhe) [inline portfolio/src/main/java/com/google/sps/data/GeneralUtility.java:20]: nit: no newline at EOF.

File: portfolio/src/main/webapp/js/comments.js [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+*/
+function getServletUrl() {
+  return document.getElementById('comment-form').classList[0];
+}
+
+/**
+* Dynamically appends comments from server to DOM.
+*/
+async function printComments() {
+
```
Comment (pengxianzhe) [inline portfolio/src/main/webapp/js/comments.js:29]: Remove the extra empty lines in this method.

File: portfolio/src/main/java/com/google/sps/data/CommentUtility.java (hunk 2 above)
Comment (mbouadeus) [reply portfolio/src/main/java/com/google/sps/data/CommentUtility.java:130]: Yes I can. Good catch.

File: portfolio/src/main/java/com/google/sps/data/GeneralUtility.java (hunk 3 above)
Comment (mbouadeus) [reply portfolio/src/main/java/com/google/sps/data/GeneralUtility.java:20]: Fixed it.

File: portfolio/src/main/java/com/google/sps/data/CommentUtility.java (hunk 1 above)
Comment (mbouadeus) [reply portfolio/src/main/java/com/google/sps/data/CommentUtility.java:129]: Fixed it.

File: portfolio/src/main/java/com/google/sps/data/CommentUtility.java (hunk 2 above)
Comment (mbouadeus) [reply portfolio/src/main/java/com/google/sps/data/CommentUtility.java:130]: Fixed it.

File: portfolio/src/main/java/com/google/sps/data/GeneralUtility.java (hunk 3 above)
Comment (mbouadeus) [reply portfolio/src/main/java/com/google/sps/data/GeneralUtility.java:20]: Fixed it.

File: portfolio/src/main/webapp/js/comments.js (hunk 4 above)
Comment (mbouadeus) [reply portfolio/src/main/webapp/js/comments.js:29]: Fixed it.

**Review by isaacwdavis — COMMENTED** [review]
LTGM

**Pull request merged**